File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/bpm/modeling/model/Process.ts
import {ProcessModel} from './ProcessModel';
import {ProcessStatus} from './ProcessStatus';

export class Process {
  id: string | undefined = undefined;
  name: string = '';
  description: string = '';
  bpmsElementId: string = '';
  version: string = '';
  processKey: string = '';
  latest: boolean = true;
  processModel: ProcessModel = new ProcessModel();
  createdAt: Date = new Date();
  lastModifiedAt: Date = new Date();
  status: ProcessStatus = ProcessStatus.ACTIVE;
  author: string = '';
  defaultPriority: number = 50;
}

File: backend/recpro-backend-lib/src/main/java/de/ubt/ai4/petter/recpro/recpro_backend_lib/recpro/model/bpm/modeling/Process.java
package de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.bpm.modeling;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Process {
    @Id
    @Builder.Default
    private String id = UUID.randomUUID().toString();

    private String name;
    private String description;
    private String bpmsElementId;
    private String version;
    private String processKey;
    private boolean latest;
    private Instant createdAt;
    private Instant lastModifiedAt;
    private String author;
    private int defaultPriority;
    @Enumerated(EnumType.STRING)
    private ProcessStatus status;

    @OneToOne(cascade = CascadeType.ALL)
    private ProcessModel processModel;
}
